kafka_consumer/datadog_checks/kafka_consumer/cluster_metadata.py

```diff
@@ -0,0 +1,641 @@
+                        }
+                    )
+
+                # Emit metrics for numeric configs
+                if 'retention.ms' in topic_config and topic_config['retention.ms'] != '-1':
+                    retention_ms = int(topic_config['retention.ms'])
+                    self.check.gauge('topic.config.retention_ms', retention_ms, tags=topic_tags)
+
+                if 'retention.bytes' in topic_config and topic_config['retention.bytes'] != '-1':
+                    retention_bytes = int(topic_config['retention.bytes'])
+                    self.check.gauge('topic.config.retention_bytes', retention_bytes, tags=topic_tags)
+
+                if 'max.message.bytes' in topic_config:
+                    max_bytes = int(topic_config['max.message.bytes'])
+                    self.check.gauge('topic.config.max_message_bytes', max_bytes, tags=topic_tags)
+
+        # Persist updated topic HWM sums with timestamp for next run
+        try:
+            snapshot = {'ts': float(now_ts), 'sums': previous_topic_hwm_sums}
+            self.check.write_persistent_cache(self.TOPIC_HWM_SUM_CACHE_KEY, json.dumps(snapshot))
+        except Exception as e:
+            self.log.debug("Could not write topic HWM cache: %s", e)
+
+    def _collect_consumer_group_metadata(self, metadata):
+        self.log.debug("Collecting consumer group metadata")
+        cluster_id = metadata.cluster_id if hasattr(metadata, 'cluster_id') else 'unknown'
+        consumer_groups_future = self.client.kafka_client.list_consumer_groups()
+        consumer_groups_result = consumer_groups_future.result(timeout=self.config._request_timeout)
+
+        if consumer_groups_result.errors:
+            self.log.warning("Errors listing consumer groups: %s", consumer_groups_result.errors)
+
+        consumer_groups = consumer_groups_result.valid
+
+        self.log.debug("Found %s consumer groups", len(consumer_groups))
+        self.check.gauge('consumer_group.count', len(consumer_groups), tags=self._get_tags(cluster_id))
+
+        group_ids = [group.group_id for group in consumer_groups]
+        if not group_ids:
+            return
+        describe_futures = self.client.kafka_client.describe_consumer_groups(group_ids)
+
+        group_id_to_info = {}
+        for group_id, future in describe_futures.items():
+            try:
+                group_id_to_info[group_id] = future.result(timeout=self.config._request_timeout)
+            except Exception as e:
+                self.log.warning("Error getting consumer group details for %s: %s", group_id, e)
+
+        for group_id, group_info in group_id_to_info.items():
+            group_tags = self._get_tags(cluster_id) + [f'consumer_group:{group_id}']
+            state = group_info.state
+            members = group_info.members
+            coordinator = group_info.coordinator
+            state_name = state.name if hasattr(state, 'name') else str(state)
+            state_tags = group_tags + [f'state:{state_name}']
```

Comment:
I'm worried that this tag is inconsistent with the consumer group state tag we already [collect here](https://github.com/DataDog/integrations-core/blob/93c63376affe5ed76d75597ef28faf242ce0dce9/kafka_consumer/datadog_checks/kafka_consumer/kafka_consumer.py#L269). I'm also worried that we're exposing this tag in 2 different ways.